casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -103,6 +94,66 @@ object Genesis {
                      )
     } yield genesisBlock
 
+  def getWallets[F[_]: Sync: Log: RaiseIOError](
+      maybeWalletsPath: Option[String],
+      defaultWalletPath: Path
+  ): F[Seq[Vault]] = {
+    def walletFromFile(walletsPath: Path): F[Seq[Vault]] =
+      for {
+        maybeLines <- SourceIO.open(walletsPath).use(_.getLines).attempt
+        wallets <- maybeLines match {
+                    case Right(lines) =>
+                      lines
+                        .traverse(fromLine(_) match {
+                          case Right(wallet) => wallet.some.pure[F]
+                          case Left(errMsg) =>
+                            Log[F]
+                              .warn(s"Error in parsing wallets file: $errMsg")
+                              .map(_ => none[Vault])
+                        })
+                        .map(_.flatten)
+                    case Left(ex) =>
+                      Log[F]
+                        .warn(
+                          s"Failed to read ${walletsPath.toAbsolutePath} for reason: ${ex.getMessage}"
+                        )
+                        .map(_ => Seq.empty[Vault])
+                  }
+      } yield wallets
+
+    maybeWalletsPath match {
+      case Some(walletsPathStr) =>
+        val walletsPath = Paths.get(walletsPathStr)
+        Monad[F].ifM(exists(walletsPath))(
+          walletFromFile(walletsPath),
+          Sync[F].raiseError(new Exception(s"Specified wallets file $walletsPath does not exist"))
+        )
+      case None =>
+        Monad[F].ifM(exists(defaultWalletPath))(
+          Log[F].info(s"Using default file $defaultWalletPath") >> walletFromFile(
+            defaultWalletPath
+          ),
+          Log[F]
+            .warn(
+              "No wallets file specified and no default file found. No wallets will exist at genesis."
```

Comment:
is this a warning situation? Isn't this a fatal error? Can RChain work without genesis wallets?

Reply:
Eventually but as of now the SRE team isn't forced to pass in wallets.txt which is convenient.

Reply:
@adaszko Is it a real problem to have the wallets.txt mandatory?

Reply:
1) If the `--wallets-file` has been passed and the file isn't usable (i.e. parsing error, the file can't be opened, etc), it's OK — and even desirable — for the node to fail
2) If the `--wallets-file` hasn't been passed and there's not wallets.txt file at the default path, the node shouldn't fail yet, as we would need to make some preparations for that case first, in integration tests, devnet and testnet. /CC @woky 

I've created a ticket for 2): https://rchain.atlassian.net/browse/RCHAIN-3573

The way to update integration tests is to prepare a PR where the `--wallets-file` option is mandatory, run integration tests on it and see what failed.  I can then patch up the problems on that branch and we'll be golden.

Reply:
ok. thanks @adaszko for clarifications